splink/files/settings_jsonschema.json

```diff
@@ -5,6 +5,7 @@
   "type": "object",
   "title": "Splink settings",
   "required": ["comparison_columns", "link_type"],
+  "additionalProperties": false,
```

Comment:
Not necessary to PR but useful - this means that if the user includes extraneous keys (e.g. a misspelt key) in the settings dictionary, they'll get a validation error